functions/src/modules/gmail.functions.ts

```diff
@@ -0,0 +1,220 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import * as Google from 'googleapis';
+import { Message } from 'firebase-functions/lib/providers/pubsub';
+import { processSQRDoneFromGmail } from './SQR.functions';
+import * as moment from 'moment';
+
+const db = admin.database();
+
+const { client_key, secret } = functions.config().gmail;
+const oauth2Client = new Google.google.auth.OAuth2(
+  client_key,
+  secret,
+  'https://us-central1-audio-seva-team-test.cloudfunctions.net/Gmail-oauth2callback'
```

Comment:
Sorry, I had not replied your question in this regard.
Please don't hard-code the particular project name into the URL, as for the production environment it will be different. I'm sure you can get the project name from Firebase.

Reply:
Yeah, that should be possible, using an environmental variable. Will check

Reply:
Using the `req.headers.host` to form the URL now.